extract/toc.js

```diff
@@ -66,6 +66,27 @@ export default function toc(tocHtml, manifest, spine) {
   };
 }
 
+function findTocRoot(currentNode, found = {}) {
+  Object.keys(currentNode).forEach(key => {
+    if (isRootNode(currentNode[key])) {
+      found = currentNode[key];
+      return found;
+    }
+    if (typeof currentNode[key] === 'object') {
+      found = findTocRoot(currentNode[key], found);
+    }
+  });
+  return found;
+}
+
+function isRootNode(node) {
+  if (typeof node !== 'undefined') {
```

Comment:
I think you can just do `if (node) {` here because you can't reassign undefined in strict mode anymore.